src/main/java/org/spongepowered/api/command/IncorrectCommandSourceException.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command;
+
+import static org.spongepowered.api.util.SpongeApiTranslationHelper.t;
+
+import org.spongepowered.api.text.Text;
+
+/**
+ * This exception is thrown when the target {@link CommandSource}s is of the
+ * correct type to call the target command.
+ */
+public class IncorrectCommandSourceException extends CommandException {
+    private static final long serialVersionUID = -2927330349931825821L;
+
+    /**
+     * Create an exception with the default message.
+     */
+    public IncorrectCommandSourceException() {
+        this(t("You cannot execute this command!"));
+    }
```

Comment:
I'm torn by this method. On the one hand I would like to send a localized message, on the other hand I would also like to specify who is allowed to execute it.

Reply:
`SpongeApiTranslationHelper` shouldn't be used anymore, translations go in the impl.

Reply:
Do bear in mind that this is a port from the previous PR and things have been missed. This will go in the impl.

I expect this message will be in two bits - one basic message and a list of allowed sources. So, something like (but not exactly):

```
You cannot execute this command using this source!
Allowable sources: Player, Console, Command Block
```